### PR TITLE
Further reduce CI runtime by having all datasets downloaded only once 

### PR DESCRIPTION
**Pull request by xuhdev**
We now download datasets to a local directory, then in most tests,
dataset downloading are from our local HTTP server.

This also helps speed up running tests locally, as downloaded datasets
are saved across test sessions.

### CODE REVIEW

**Comment by xuhdev** [comment]
It doesn't seem have much effects though, likely because I couldn't change some downloading spots because pydax doesn't allow customizing default schema file URLs. But the framework should be good for future changes. I've opened an issue for customizing default schema file URLs.

File: .travis.yml [hunk 1]
```diff
@@ -57,6 +57,10 @@ branches:
   only:
     - master
 
+cache:
+  directories:
+    - /home/travis/tests/datasets
```
Comment (edwardleardi) [inline .travis.yml]:
```suggestion
    - $HOME/tests/datasets
```
Should we use the Travis default env var here? https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

**Review by edwardleardi — DISMISSED** [review]
Added one suggestion. Very interesting usage of pytest, learned a lot from this PR

File: .travis.yml (hunk 1 above)
Comment (edwardleardi) [reply .travis.yml]: Guess this doesn't matter actually since we aren't using Travis anymore

**Review by edwardleardi — APPROVED** [review]

**Pull request merged**